checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsTransfer.java

```diff
@@ -24,15 +50,83 @@ public CalledMethodsTransfer(final CFAnalysis analysis) {
           ((CalledMethodsAnnotatedTypeFactory) atypeFactory)
               .adjustMethodNameUsingValueChecker(methodName, node.getTree());
-      accumulate(receiver, result, methodName);
+      accumulate(receiver, superResult, methodName);
+    }
+    TransferResult<CFValue, CFStore> finalResult =
+        new ConditionalTransferResult<>(
+            superResult.getResultValue(),
+            superResult.getThenStore(),
+            superResult.getElseStore(),
+            exceptionalStores);
+    exceptionalStores = null;
+    return finalResult;
+  }
+
+  @Override
+  public void accumulate(Node node, TransferResult<CFValue, CFStore> result, String... values) {
+    super.accumulate(node, result, values);
+    if (exceptionalStores == null) {
+      return;
+    }
+
+    List<String> valuesAsList = Arrays.asList(values);
+    JavaExpression target = JavaExpression.fromNode(node);
+    if (CFAbstractStore.canInsertJavaExpression(target)) {
+      CFValue flowValue = result.getRegularStore().getValue(target);
+      if (flowValue != null) {
+        // Dataflow has already recorded information about the target.  Integrate it into the list
+        // of values in the new annotation.
+        Set<AnnotationMirror> flowAnnos = flowValue.getAnnotations();
+        assert flowAnnos.size() <= 1;
+        for (AnnotationMirror anno : flowAnnos) {
+          if (atypeFactory.isAccumulatorAnnotation(anno)) {
+            List<String> oldFlowValues =
+                AnnotationUtils.getElementValueArray(
+                    anno,
+                    ((CalledMethodsAnnotatedTypeFactory) atypeFactory).calledMethodsValueElement,
+                    String.class);
+            // valuesAsList cannot have its length changed -- it is backed by an
+            // array.  getElementValueArray returns a new, modifiable list.
+            oldFlowValues.addAll(valuesAsList);
+            valuesAsList = oldFlowValues;
+          }
+        }
+      }
+      AnnotationMirror newAnno = atypeFactory.createAccumulatorAnnotation(valuesAsList);
+      exceptionalStores.forEach((tm, s) -> s.insertValue(target, newAnno));
+    }
+  }
+
+  /**
+   * Create a set of stores for the exceptional paths out of the block containing {@code node}. This
+   * allows propagation, along those paths, of the fact that the method being invoked in {@code
+   * node} was definitely called.
+   *
+   * @param node a method invocation
+   * @param input the transfer input associated with the method invocation
+   * @return a map from types to stores. The keys are the same keys used by {@link
+   *     ExceptionBlock#getExceptionalSuccessors()}. The values are copies of the regular store from
+   *     {@code input}.
+   */
+  private Map<TypeMirror, CFStore> makeExceptionalStores(
+      MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+    if (!(node.getBlock() instanceof ExceptionBlock)) {
+      // this can happen in some weird (buggy?) cases
```

Comment:
If you like we can link to the relevant Checker Framework issue here, which I opened some time back: https://github.com/typetools/checker-framework/issues/3585